calliope/core/preprocess/checks.py

```diff
@@ -776,4 +776,23 @@ def check_model_data(model_data):
             'MILP constraints.'
         )
 
+    # Check for storage_initial being greater than or equal to the storage_dod
+
+    if hasattr(model_data, 'loc_techs_storage'):
+        for loc_tech in model_data.loc_techs_storage.values:
+            if hasattr(model_data, 'storage_initial') and hasattr(model_data, 'storage_dod'):
+                if (model_data.storage_initial.loc[{'loc_techs_store': loc_tech}].values
+                        < model_data.storage_dod.loc[{'loc_techs_store': loc_tech}].values):
+                    model_data.storage_initial.loc[{'loc_techs_store': loc_tech}] = model_data.storage_dod.loc[{'loc_techs_store': loc_tech}]
+                    model_warnings.append(
+                        'storage_initial is smaller than storage_dod.'
```

Comment:
It does lead to infeasibility, because the new storage_dod constraint forces the storage state of charge to be always greater than or equal to the storage_dod, which is violated at timestep_0 if storage_initial is lower than storage_dod. There's no way for the model to charge the storage at timestep_0, and the two constraints conflict without any possible solution, which leads to infeasibility.

Reply:
That's not why it would be infeasible, because the constraint is:

``backend_model.storage[loc_tech, timestep] == storage_previous_step - carrier_prod - carrier_con``
where `storage_previous_step` is equal to `storage_initial` in timestep 0. This would allow a model to have non-zero `carrier_con` in timestep 0 such that `storage` reaches a value equal to, or greater than, the depth of discharge. It would be good to get to the bottom of why it leads to infeasibility, if it isn't this constraint...

Reply:
Yep, the problem is that the carrier_con that can occur in a single timestep is very often not infinite, but rather limited by either charge_rate or by competing energy needs (e.g. need to satisfy a demand consumes carrier that could otherwise be stored). And so, very often this leads to infeasibility. If you create conditions for which the gap between storage initial and dod can be covered in a single timestep, than it is fine. But I really think that the easiest thing would be to say: if you define a storage_dod than the storage_initial must be at least equal to that, or greater.

And here comes an issue, related to your other comment about using fractional values rather than absolute ones: I totally agree that a fractional value is much better, because if you run in "plan" mode there's no way to use an absolute value properly. The point is: why isn't storage_initial a fractional value as well?? I guess it should also be fractional to be used properly in "plan" mode.
And making both of them fractional would make possible to keep all the tests and checks I proposed. Otherwise if one is fractional and the other is absolute is quite a mess.

What do you think?